activestorage/app/controllers/active_storage/blobs/proxy_controller.rb

```diff
@@ -5,8 +5,17 @@ class ActiveStorage::Blobs::ProxyController < ActiveStorage::BaseController
   include ActiveStorage::SetBlob
 
   def show
```

Comment:
Just to understand the reasoning behind it. Why is this one the only default controller changed? I mean, there are 4 controllers for serving files/variations:
`activestorage/app/controllers/active_storage/blobs/proxy_controller.rb`
`activestorage/app/controllers/active_storage/blobs/redirect_controller.rb`
`activestorage/app/controllers/active_storage/representations/proxy_controller.rb`
`activestorage/app/controllers/active_storage/representations/redirect_controller.rb`

I'm not sure if keep adding logic to default controllers or if it'd be possible to achieve this in some other way. For example, shouldn't app users who want this feature create a new route + controller to implement the feature? Or why do you think this should be available(working) for everyone by default?

Reply:
That's a fair point, thanks for taking a look!

I think byte range support is common enough that is almost assumed to already be supported. Each of the underlying services even utilize it through the `download_chunk` method. So with that it'd make sense to bake it into the proxy controllers.

With the redirect controllers, I am a lot less familiar. I think it is already supported because we'd pass back the url and the client would still pass the byte range header to the redirected url.

If all that makes sense to you, I will update `activestorage/app/controllers/active_storage/representations/proxy_controller.rb` to also support the byte range requests

Reply:
>With the redirect controllers, I am a lot less familiar. I think it is already supported because we'd pass back the url and the client would still pass the byte range header to the redirected url.

In this case, wouldn't it make sense to keep the proxy controllers simpler and use the redirect urls to implement this feature in an app?

>If all that makes sense to you, I will update activestorage/app/controllers/active_storage/representations/proxy_controller.rb to also support the byte range requests

I think it makes sense to have the representations proxy controller match the blobs proxy controller behavior and maybe it'd be cool to have it extracted into a controller concern. But, just to set clear expectations, I don't have a say on what can be merged or not in Rails 😄 I'm giving feedback just to try to help PRs move forward, but then it's up to some Rails core/committer member to decide whether this is a good idea or not.

Reply:
Sorry it's taken so long to get back to you, I wanted to dive into the code a little bit so I can answer better.

> In this case, wouldn't it make sense to keep the proxy controllers simpler and use the redirect urls to implement this feature in an app?

I think it'd be good to support it as much as possible. I don't think someone should have to use redirects just to get this support if proxy supports their use case better for everything else. 

> I think it makes sense to have the representations proxy controller match the blobs proxy controller behavior and maybe it'd be cool to have it extracted into a controller concern.

Totally agree with this. Most of it is in a concern already! Tonight I started updating representations to use the concern, and it was super simple until... It looks like we don't have the `byte_size` for the variants. Without the size we can't tell if the byte ranges are valid or pass the proper headers back. I'm kinda stumped here since I don't quite know how/where we could save them

> But, just to set clear expectations, I don't have a say on what can be merged or not in Rails 😄 I'm giving feedback just to try to help PRs move forward, but then it's up to some Rails core/committer member to decide whether this is a good idea or not.

I appreciate you doing this!